### PR TITLE
Add WebSocket 2023 Channel Type

### PR DESCRIPTION
**Pull request by acoburn**
This is a first pass at a WebSocket Channel type for 2023. I need to go over all of this will a bit more focus, but it is mostly ready now.

### CODE REVIEW

File: websocket-channel-2023.html [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+<code>Authorization: DPoP &lt;token&gt;</code>
+<code>DPoP: &lt;proof&gt;</code>
+<code>Content-Type: application/ld+json</code>
+<code></code>
+<code>{</code>
+<code>  "@context": ["https://www.w3.org/ns/solid/notification/v1"],</code>
+<code>  "type": "WebSocketChannel2023",</code>
+<code>  "topic": "https://storage.example/resource",</code>
+<code>  "state": "opaque-state",</code>
+<code>  "expiration": "2022-12-27T12:37:15Z",</code>
```
Comment (csarven) [inline websocket-channel-2023.html:398]:
```suggestion
<code>  "endAt": "2022-12-27T12:37:15Z",</code>
```

**Review by csarven — APPROVED** [review]
Pass through content and identifiers including "subscription" would be good, including all references to `TR/notifications-protocol` (expected based on ED->PR) or perhaps references to `TR/{dddd}/notifications-protocol-yyyymmdd` if so desired. Although we don't have that identifier at this moment, I can set the date to be published / PR'd at 2022-12-31 or 2023-01-01 if it makes minor sense to align with 2023). We can have notifications-protocol and websocket-channel-2023 have the same yyyymmdd.

**Comment by csarven** [comment]
https://github.com/solid/notifications/issues/74
https://github.com/solid/vocab/pull/62

**Review by elf-pavlik — APPROVED** [review]

File: websocket-channel-2023.html [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+<code>ws.onmessage(evt => console.log("Message received: ", evt))</code></pre>
+
+                    <figcaption property="schema:name">JavaScript code to handle <code>onclose</code> and <code>onmessage</code> events</figcaption>
+                  </figure>
+                </div>
+              </section>
+            </div>
+          </section>
+
+          <section id="authentication-authorization" inlist="" rel="schema:hasPart" resource="#authentication-authorization">
```
Comment (csarven) [inline websocket-channel-2023.html:448]: Unless there are different requirements, this section should match/refer to notifications-protocol's authentication-authorization

File: websocket-channel-2023.html [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+
+              <p>A client establishes a subscription using the <code>WebSocketChannel2023</code> type by sending an authorized subscription request to the hypermedia API retrieved via <cite><a href="https://solid.github.io/notifications/protocol#discovery">Solid Notifications discovery</a></cite>.</p>
+
+              <p>For <code>WebSocketChannel2023</code> interactions, the client sends a JSON-LD payload to the appropriate hypermedia API via <code>POST</code>. The only required fields in this interaction are <code>type</code> and <code>topic</code>. <span about="" id="client-subcription-type" rel="spec:requirement" resource="#client-subscription-type"><span property="spec:statement">The <code>type</code> field <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> contain the type of subscription being requested.</span></span> <span about="" id="client-subscription-feature" rel="spec:requirement" resource="#client-subscription-feature"><span property="spec:statement">The <code>topic</code> field <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> contain the URL of the resource a client wishes to subscribe to changes.</span></span></p>
```
Comment (CxRes) [inline websocket-channel-2023.html:376]: Replace:
MUST</span> contain the URL of the resource a client wishes to subscribe to changes.</span></span></p>

With:
MUST</span> contain the URL of the resources a client wishes to subscribe to changes.</span></span></p>

**Pull request merged**